src/core/lib/security/credentials/tls/grpc_tls_credentials_options.h

```diff
@@ -126,6 +126,25 @@ struct grpc_tls_credentials_options
   // not enable CRL checking. Only supported for OpenSSL version > 1.1.
   void set_crl_directory(std::string path) { crl_directory_ = std::move(path); }
 
+  bool operator==(const grpc_tls_credentials_options& other) const {
```

Comment:
One down-side of this approach is that any time we add a new field to `grpc_tls_credentials_options`, we need to remember to change `operator==()` to look at the new field.  Is there some way of automating this such that that happens automatically?  If not, we could be opening ourselves up to a security problem in the future.

Reply:
We get default comparison operations in C++20 :)

Reply:
Sure, but we have no idea when we'll upgrade to C++20, so that doesn't solve the problem for now.  Between now and then, I don't want to be in a situation where forgetting to add a field will cause a security problem.  I think we need some automation to address this.

For example, maybe we can write a small script to generate this file from input that defines each field name and type, so it can automatically generate the comparator method.  The script could also auto-generate the test to verify that the comparison fails when each field's value differs.

Reply:
Done

Reply:
Wow the script is super cool. Thanks for doing that!

Reply:
This looks great!

Please also add these generated files to the [`.gitattributes`](https://github.com/grpc/grpc/blob/master/.gitattributes) file so that they don't show up in diffs by default.

Also, please make sure this script is automatically run as part of the sanity check before PR submission, so that people don't accidentally manually edit these files and not notice that they actually need to do it via the script.